core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -294,6 +294,33 @@ public static boolean containsNullable(RelDataType type) {
     return false;
   }
 
+  /**
+   * Recreates a RelDataType with the type keep same as the param sourceRelDataType and
+   * nullability keep same as the param targetRelDataType.
+   */
+  public static RelDataType keepNullableSame(RelDataType sourceRelDataType,
```

Comment:
Picking a good name for this method seems hard, what about `keepSourceTypeAndTargetNullability`?

Reply:
Looks good, I will fix it.